__tests__/app.test.js

```diff
@@ -84,82 +84,165 @@ describe("GET", () => {
-      .then((response) => {
-        expect(response.body.msg).toBe('Bad Request')
-      })
-    })
+        .get("/api/articles/not-an-id/comments")
+        .expect(400)
+        .then((response) => {
+          expect(response.body.msg).toBe("Bad Request");
+        });
+    });
   });
-  
 });
 
+//========================POST =========================
 
+describe("POST", () => {
+  describe("/api/articles/:article_id/comments", () => {
+    test("STATUS 201 and should insert a new comment with the article_id and return the comment back to the client", () => {
+      const newComment = {
+        username: "rogersop",
+        body: "Lobster pot",
+      };
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+          const { Newcomment } = response.body;
+          const timeCreated_at = Newcomment[0]["created_at"];
+          expect(Newcomment[0]).toEqual({
+            comment_id: 19,
+            body: "Lobster pot",
+            article_id: 3,
+            author: "rogersop",
+            votes: 0,
+            created_at: timeCreated_at,
+          });
+        });
+    });
+    describe("/api/articles/999/comments", () => {
+      test("STATUS 404 and should respond with a message when given an valid parametric endpoint but does not exist", () => {
+        const newComment = {
+          username: "rogersop",
+          body: "Lobster pot",
+        };
+        return request(app)
+          .post("/api/articles/999/comments")
+          .send(newComment)
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe("Not Found");
+          });
+      });
+    });
+    describe("/api/articles/ananas/comments", () => {
+      test("STATUS 400 and should respond with a message of Bad request when given an invalid parametric endpoint", () => {
+        const newComment = {
+          username: "rogersop",
+          body: "Lobster pot",
+        };
+        return request(app)
+          .post("/api/articles/ananas/comments")
+          .send(newComment)
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe("Bad Request");
+          });
+      });
+    });
+    describe("/api/articles/:article_id/comments - POST comment with no keys", () => {
+      test("STATUS 400 and should respond with a message of Bad request when sent a body/comment with no keys", () => {
+        const newComment = {}
+        return request(app)
+          .post("/api/articles/3/comments")
+          .send(newComment)
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe("Bad Request");
+          });
+      });
+    });
+    describe("/api/articles/:article_id/comments - POST comment with invalid username", () => {
+      test("STATUS 400 and should respond with a appropriate message when username is invalid", () => {
```

Comment:
I can see why you have gone for a 400 status here but it could be argues that the user is "not found" so should be a 404.

Reply:
Changed it to 404 again just after I submitted  :)